File: memory-domain/src/main/java/com/memory/domain/game/repository/GameSessionRepositoryCustom.java
package com.memory.domain.game.repository;

import com.memory.domain.game.GameMode;
import com.memory.domain.game.GameSession;
import com.memory.domain.game.GameSessionStatus;
import com.memory.domain.member.Member;

import java.util.List;
import java.util.Optional;

public interface GameSessionRepositoryCustom {

    Optional<GameSession> findGameSessionById(Long sessionId);

    Optional<GameSession> findByMemberAndStatus(Member member, GameSessionStatus gameSessionStatus);

    List<GameSession> findByMemberAndGameMode(Member member, GameMode gameMode, Long lastSessionId, Integer size);

}

File: memory-domain/src/main/kotlin/com/memory/domain/game/repository/GameSessionRepositoryCustom.kt
package com.memory.domain.game.repository

import com.memory.domain.game.GameMode
import com.memory.domain.game.GameSession
import com.memory.domain.game.GameSessionStatus
import com.memory.domain.member.Member
import java.util.Optional

interface GameSessionRepositoryCustom {

    fun findGameSessionById(sessionId: Long?): Optional<GameSession>

    fun findByMemberAndStatus(member: Member?, gameSessionStatus: GameSessionStatus?): Optional<GameSession>

    fun findByMemberAndGameMode(member: Member?, gameMode: GameMode?, lastSessionId: Long?, size: Int?): List<GameSession>

}
